source/PlayerInfo.cpp

```diff
@@ -2110,6 +2112,55 @@ void PlayerInfo::SetTravelDestination(const Planet *planet)
 
 
 
+void PlayerInfo::QueueTeleport(const Planet *destination, bool flagshipOnly)
+{
+	teleportPlanet = destination;
+	flagshipOnly = flagshipOnly;
+}
+
+
+
+void PlayerInfo::DoQueuedTeleport()
+{
+	if(!teleportPlanet || !FlagshipPtr())
+		return;
+	if(!planet)
+	{
+		teleportPlanet = nullptr;
+		return;
+	}
+
+	flagship->SetSystem(teleportPlanet->GetSystem());
+	flagship->SetPlanet(teleportPlanet);
+	if(!flagshipOnly)
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsParked() && !ship->IsDestroyed() && ship->GetPlanet() == planet)
```

Comment:
We're moving the flagship twice here I think
Not that I really mind it though its prob better than adding a check for all ships I guess